src/components/checkbox/index.js

```diff
@@ -53,46 +53,34 @@ const Label = styled.label`
   `};
 `
 
-class Checkbox extends Component {
-  constructor(props) {
-    super(props)
+const Checkbox = ({ checked, handleChange, id, value, name, label }) => {
+  const [checkedState, setCheckedState] = useState(checked)
 
-    this.state = {
-      checked: this.props.checked,
-    }
-  }
+  useEffect(() => {
+    setCheckedState(checked)
+  }, [checked])
 
-  static getDerivedStateFromProps(nextProps) {
-    return { checked: nextProps.checked }
-  }
+  const toggleCheckbox = e => {
+    setCheckedState(e.target.checked)
```

Comment:
What would be ideal here would be to take the concept of state out of the component, and into a wrapper component (which could be in the same file). The `Checkbox` itself should be "dumb" and only have the concept of an `onCheckedHandler` function, which is called on un/checking the `Input` field. This will also make testing easier, as the dumb component could be tested separately.

Reply:
This would also help us to move towards our Storybook goal. De-coupling the presentational components from the stateful.

Reply:
I've revised my code but I think I'm not quite on the mark with what you intended, take a look and add a description to the ticket, it'll be good to think about it for the other tickets too.

Reply:
Ok, just leave it as part of this ticket. Parity is fine for now. We'll do the de-couple later.